source/common/router/rds_impl.cc

```diff
@@ -108,6 +111,8 @@ void RdsRouteConfigSubscription::onConfigUpdate(
     if (config_update_info_->routeConfiguration().has_vhds()) {
       ENVOY_LOG(debug, "rds: vhds configuration present, starting vhds: config_name={} hash={}",
                 route_config_name_, config_update_info_->configHash());
+      // TODO(dmitri-d): It's unsafe to depend directly on factory context here,
```

Comment:
@dmitri-d I tagged you here. The use of factory context outside of the constructor is probably unsafe, since the subscription might outlive the listener (at least that's what ASAN tells me).